app/Models/Album.php

```diff
@@ -150,44 +151,32 @@ public function toArray(): array
 		return $result;
 	}
 
-	public function delete(): bool
+	/**
+	 * {@inheritDoc}
+	 */
+	public function performDeleteOnModel(): void
 	{
-		try {
-			$this->refreshNode();
-
-			$success = true;
-
-			// Delete all recursive child photos first
-			$photos = $this->all_photos()->lazy();
-			/** @var Photo $photo */
-			foreach ($photos as $photo) {
-				// This also takes care of proper deletion of physical files from disk
-				// Note, we need this strange condition, because `delete` may also
-				// return `null` on success, so we must explicitly test for
-				// _not `false`_.
-				$success &= ($photo->delete() !== false);
-			}
-
-			if (!$success) {
-				return false;
-			}
-
-			// Finally, delete the album itself
-			// Note, we need this strange condition, because `delete` may also
-			// return `null` on success, so we must explicitly test for
-			// _not `false`_.
-			$success &= (parent::delete() !== false);
-
-			return $success;
-		} catch (\Exception $e) {
-			try {
-				// if anything goes wrong, don't leave the tree in an inconsistent state
-				$this->newModelQuery()->fixTree();
-			} catch (\Throwable) {
-				// Sic! We cannot do anything about the inner exception
-			}
-			throw $e;
-		}
+		$fileDeleter = (new Delete())->do([$this->id]);
```

Comment:
```suggestion
		$albumDeleter = (new Delete())->do([$this->id]);
```

Reply:
No. `FileDeleter` is correct, because `->do` returns an object which delete files. The albums have already been deleted from the DB.